sqlmesh/core/console.py

```diff
@@ -508,159 +549,200 @@ def _show_summary_tree_for(
+                )
             tree.add(removed_tree)
-        if modified_names:
+        if modified_snapshot_ids:
             direct = Tree(f"[bold][direct]Directly Modified:")
             indirect = Tree(f"[bold][indirect]Indirectly Modified:")
             metadata = Tree(f"[bold][metadata]Metadata Updated:")
-            for name in modified_names:
+            for s_id in modified_snapshot_ids:
+                name = s_id.name
+                display_name = s_id.display_name(environment_naming_info, default_catalog)
                 if context_diff.directly_modified(name):
                     direct.add(
-                        f"[direct]{name}"
+                        f"[direct]{display_name}"
                         if no_diff
-                        else Syntax(
-                            f"{name}\n{context_diff.text_diff(name)}", "sql", word_wrap=True
-                        )
+                        else Syntax(f"{display_name}\n{context_diff.text_diff(name)}", "sql")
                     )
                 elif context_diff.indirectly_modified(name):
-                    indirect.add(f"[indirect]{name}")
+                    indirect.add(f"[indirect]{display_name}")
                 elif context_diff.metadata_updated(name):
                     metadata.add(
-                        f"[metadata]{name}"
+                        f"[metadata]{display_name}"
                         if no_diff
-                        else Syntax(
-                            f"{name}\n{context_diff.text_diff(name)}", "sql", word_wrap=True
-                        )
+                        else Syntax(f"{display_name}", "sql", word_wrap=True)
                     )
             if direct.children:
                 tree.add(direct)
             if indirect.children:
                 tree.add(indirect)
             if metadata.children:
                 tree.add(metadata)
-        if selected_ignored_names:
-            tree.add(self._get_ignored_tree(selected_ignored_names, selected_snapshots))
+        if selected_ignored_snapshot_ids:
+            tree.add(
+                self._get_ignored_tree(
+                    selected_ignored_snapshot_ids,
+                    selected_snapshots,
+                    environment_naming_info,
+                    default_catalog,
+                )
+            )
         self._print(tree)
 
-    def _show_options_after_categorization(self, plan: Plan, auto_apply: bool) -> None:
+    def _show_options_after_categorization(
+        self, plan: Plan, auto_apply: bool, default_catalog: t.Optional[str]
+    ) -> None:
         if plan.forward_only and plan.new_snapshots:
-            self._prompt_effective_from(plan, auto_apply)
+            self._prompt_effective_from(plan, auto_apply, default_catalog)
 
         if plan.requires_backfill:
-            self._show_missing_dates(plan)
-            self._prompt_backfill(plan, auto_apply)
+            self._show_missing_dates(plan, default_catalog)
```

Comment:
I actually started with this approach but didn't like it. There are places where we call console methods that don't require default catalog because we aren't printing the status of individual models. These places also didn't have a default catalog already since it wasn't contextually relevant. Therefore, instead of passing it around to places who didn't need it, I decided to make it something you pass in instead when you do need it.

Reply:
Btw I just realized that `display_name` should probably be cached since it gets called a lot. I will add that.

Reply:
it can just default to None when we don't need it right?

Reply:
Then you could construct consoles that are capable of properly printing model names and some consoles that couldn't. I find that state worse then just passing in default catalog for methods that need it.

Reply:
i disagree

Reply:
This is a minor detail. It can be refactored in a follow up PR if that is what you would prefer. I would just ask that if we do go this route that methods that expect a default catalog verify it is defined and raised if it is not so we don't silently print bad model names.

Reply:
sure no problem